File: RokidSDKDemoJava/app/src/main/java/com/rokid/mobile/sdk/demo/java/device/DeviceIndexFragment.java
package com.rokid.mobile.sdk.demo.java.device;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.view.View;
import android.view.ViewGroup;

import com.rokid.mobile.sdk.demo.java.R;
import com.rokid.mobile.sdk.demo.java.base.BaseFragmentAdapter;
import com.rokid.mobile.sdk.demo.java.base.BaseFragment;
import com.rokid.mobile.sdk.demo.java.base.BaseFragmentPresenter;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;

/**
 * Created by tt on 2018/2/28.
 */

public class DeviceIndexFragment extends BaseFragment {

    @BindView(R.id.device_tab_layout)
    TabLayout deviceTab;

    @BindView(R.id.device_viewPager)
    ViewPager deviceVp;

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_device_index;
    }

    @Override
    protected BaseFragmentPresenter initPresenter() {
        return null;
    }

    @Override
    protected void initVariables(View rootView, ViewGroup container, @Nullable Bundle savedInstanceState) {
        deviceTab.setupWithViewPager(deviceVp);
        List<BaseFragmentAdapter.Node> deviceList = new ArrayList<>();
        deviceList.add(new BaseFragmentAdapter.Node("绑定", new DeviceBindFragment()));
        deviceList.add(new BaseFragmentAdapter.Node("设备列表", new DeviceListFragment()));
        BaseFragmentAdapter mAdapter = new BaseFragmentAdapter(getFragmentManager(), deviceList);

        deviceVp.setAdapter(mAdapter);
    }

    @Override
    protected void initListeners() {
    }
}

File: RokidSDKDemoKotlin/app/src/main/kotlin/com/rokid/mobile/sdk/demo/device/DeviceIndexFragment.kt
package com.rokid.mobile.sdk.demo.device

import com.rokid.mobile.sdk.demo.R
import com.rokid.mobile.sdk.demo.base.BaseActivity
import com.rokid.mobile.sdk.demo.base.BaseFragment
import com.rokid.mobile.sdk.demo.base.BaseFragmentAdapter
import kotlinx.android.synthetic.main.device_fragment_index.view.*

/**
 * Description: TODO
 * Author: Shper
 * Version: V0.1 2018/2/26
 */
class DeviceIndexFragment : BaseFragment() {

    override fun layoutId(): Int = R.layout.device_fragment_index

    override fun initViews() {
        rootView!!.device_title.text = "设备"
        rootView!!.device_subtitle.text = "SDK Device Demo"

        rootView!!.device_tab_layout.setupWithViewPager(rootView!!.device_viewPager)
        rootView!!.device_viewPager.adapter = BaseFragmentAdapter(activity.supportFragmentManager,
                DeviceData.fragmentList)
    }

    override fun initListeners() {
    }

    object DeviceData {
        val fragmentList = listOf(
                BaseFragmentAdapter.Companion.Node("绑定", DeviceBinderFragment()),
                BaseFragmentAdapter.Companion.Node("设备列表", DeviceListFragment())
        )
    }

}